pkgs/tools/security/sha1collisiondetection/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchFromGitHub, libtool }:
+
+stdenv.mkDerivation  rec {
+  pname = "sha1collisiondetection";
+  version = "git-20170221";
```

Comment:
@joachifm : sha1sum is broken, because it implements a hashing algorithm that is now known to be broken. This tool implements exactly that same algorithm, but as it calculates the hash it *also* does a thorough check on whether the object for which the hash is requested has not been manipulated as described in the paper by Stevens et al. In the latter case, it will detect this, and the result is *another* SHA1 - meaning the user is alarmed and the attack has failed. 

My argument is thus, that there is no reason to keep the sha1sum as it is in NixOS - its only function can be dealt with by this tool better (the new version is twice as fast). And the new tool has the advantage of protection against a core function of the tool, namely using hashes to protect the integrity of objects from the internet. 

So I'd rename sha1sum into sha1sum.legacy and symlink the new tool to sha1sum. This is partially orthogonal to addressing the presence of collisions (all other uses of sha1sum by users of NixOS should benefit as well), if it weren't for the fac that it actually *does* mitigate the risk of having 8000+ hashes still in place. These should be safe against this collision attack.